tests/Unit/models/vpn/ipsec/ipsec-prefix.model.spec.ts

```diff
@@ -0,0 +1,1281 @@
+      }
+    });
+
+    it('should handle database errors', async () => {
+      const req: any = {
+        dbCon: db.getQuery(),
+        body: {
+          name: 'Test-Prefix'.repeat(1000), // Intentionally long name to trigger error
+          ipsec: fwcloudProduct.ipsecServer.id,
+        },
+      };
+
+      try {
+        await IPSecPrefix.createPrefix(req);
+      } catch (error) {
+        expect(error).to.exist;
+        expect(error.message).to.include('Data too long');
+      }
+    });
+  });
+
+  describe('modifyPrefix', () => {
+    it('should modify an existing prefix', async () => {
+      const req: any = {
+        dbCon: db.getQuery(),
+        body: {
+          prefix: fwcloudProduct.ipsecPrefix.id,
+          name: 'Modified-Prefix',
+        },
+      };
+
+      await IPSecPrefix.modifyPrefix(req);
+
+      const result = await db
+        .getSource()
+        .getRepository(IPSecPrefix)
+        .findOne({
+          where: { id: fwcloudProduct.ipsecPrefix.id },
+        });
+
+      expect(result).to.exist;
+      expect(result.name).to.equal('Modified-Prefix');
+    });
+
+    it('should handle database errors', async () => {
+      const req: any = {
+        dbCon: db.getQuery(),
+        body: {
+          prefix: fwcloudProduct.ipsecPrefix.id,
+          name: 'Modified-Prefix'.repeat(1000), // Intentionally long name to trigger error
+        },
+      };
+
+      try {
+        await IPSecPrefix.modifyPrefix(req);
+      } catch (error) {
+        expect(error).to.exist;
+        expect(error.message).to.include('Data too long');
+      }
+    });
+  });
+
+  describe('deletePrefix', () => {
+    it('should delete an existing prefix', async () => {
+      await IPSecPrefix.deletePrefix(db.getQuery(), fwcloudProduct.ipsecPrefix.id);
+
+      const result = await db
+        .getSource()
+        .getRepository(IPSecPrefix)
+        .findOne({
+          where: { id: fwcloudProduct.ipsecPrefix.id },
+        });
+
+      expect(result).to.not.exist;
+    });
+
+    it('should not fail when deleting a non-existent prefix', async () => {
+      const nonExistentId = -9999; // Non-existent prefix ID
+      await IPSecPrefix.deletePrefix(db.getQuery(), nonExistentId);
+    });
+  });
+
+  describe('deletePrefixAll', () => {
+    it('should delete all prefixes under a firewall', async () => {
+      const sql = `SELECT PRE.* FROM ipsec_prefix as PRE
+        INNER JOIN ipsec VPN ON VPN.id = PRE.ipsec
+        INNER JOIN firewall FW ON FW.id = VPN.firewall
+        WHERE FW.id = ${fwcloudProduct.firewall.id} AND FW.fwcloud = ${fwcloudProduct.fwcloud.id}`;
+
+      const result = await IPSecPrefix.deletePrefixAll(
+        db.getQuery(),
+        fwcloudProduct.fwcloud.id,
+        fwcloudProduct.firewall.id,
+      );
+
+      expect(result).to.exist;
+
+      const resultAfter = await db.getSource().query(sql);
+
+      expect(resultAfter).to.be.an('array').that.is.empty;
+    });
+  });
```

Comment:
The test does not validate the result returned by the method (for example, `affectedRows`).